test/gateway/eip20_gateway/confirm_redeem_intent.js

```diff
@@ -0,0 +1,494 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const EIP20Gateway = artifacts.require('TestEIP20Gateway.sol'),
+  Token = artifacts.require("MockUtilityToken"),
+  Utils = require('../../test_lib/utils'),
+  MockToken = artifacts.require("MockToken"),
+  BN = require('bn.js'),
+  TestDataJSON = require('./test_data/redeem_progressed_1'),
+  TestDataJSON2 = require('./test_data/redeem_progressed_2'),
+  messageBus = require('../../test_lib/message_bus.js'),
+  MockOrganization = artifacts.require('MockOrganization.sol'),
+  EventDecoder = require('../../test_lib/event_decoder.js');
+
+let MessageStatusEnum = messageBus.MessageStatusEnum;
+
+contract('EIP20Gateway.confirmRedeemIntent() ', function (accounts) {
+
+  let utilityToken,
+    bountyAmount,
+    coreAddress,
+    owner,
+    burnerAddress,
+    dummyStateRootProvider,
+    eip20Gateway,
+    mockToken,
+    baseToken,
+    mockOrganization,
+    worker,
+    redeemRequest,
+    redeemRequestWithNonceTwo,
+    messageHash = TestDataJSON.gateway.confirm_redeem_intent.return_value.returned_value.messageHash_;
+
+  async function confirmRedeemIntent(stubData) {
+
+    await eip20Gateway.confirmRedeemIntent(
+      stubData.redeemer,
+      stubData.nonce,
+      stubData.beneficiary,
+      stubData.amount,
+      stubData.gasPrice,
+      stubData.gasLimit,
+      stubData.blockNumber,
+      stubData.hashLock,
+      stubData.storageProof,
+    );
+  }
+
+  beforeEach(async function () {
+
+      mockToken = await MockToken.new({from: accounts[0]});
+      baseToken = await MockToken.new({from: accounts[0]});
+
+      redeemRequest = TestDataJSON.gateway.confirm_redeem_intent.params;
```

Comment:
Can we rename `redeemRequest` to some local variable? It gets confusing in the below line.